multi-node/aws/pkg/config/config.go

```diff
@@ -104,7 +107,16 @@ const (
 
 func (c Cluster) Config() (*Config, error) {
 	config := Config{Cluster: c}
-	config.ETCDEndpoints = fmt.Sprintf("http://%s:2379", c.ControllerIP)
```

Comment:
this whole thing should probably just take a full url, not just an IP. Being able to specify different ports, and http vrs https is important, so I think you probably should only retain the `etcdEndpoints := strings.Split(c.EtcdEndpoints, ",")` and then parse each URL to make sure it's valid, but not change anything.

Something like:
```
etcdEndpoints := strings.Split(c.EtcdEndpoints, ",")
for i, ep := range etcdEndpoints {
    // Check for a valid url, but leave it as-is otherwise.
    _, err := url.Parse(ep); 
    if err != nil {
        return nil, err
    }

}
config.EtcdEndpoint = etcdEndpoints[0]
config.EtcdEndpoints = c.EtcdEndpoints
```

Then below for the `curl` it could just be
```
{{ .EtcdEndpoint }}/v2/keys/coreos.com/network/config?prevExist=false
```